lib/ts/framework/h3/framework.ts

```diff
@@ -0,0 +1,201 @@
+            throw new Error("Error while trying to get original url");
+        }
+        return path;
+    };
+}
+
+export class H3Response extends BaseResponse {
+    private event: H3Event;
+    private statusCode: number;
+
+    constructor(event: H3Event) {
+        super();
+        this.original = event;
+        this.event = event;
+        this.statusCode = 200;
+    }
+
+    sendHTMLResponse = (html: string) => {
+        if (this.event.node.res.writable) {
+            this.event.node.res.setHeader("Content-Type", "text/html");
+            this.event.node.res.statusCode = this.statusCode;
+            send(this.event, html);
+            // this.response.status(this.statusCode).send(Buffer.from(html));
+        }
+    };
+
+    setHeader = (key: string, value: string, allowDuplicateKey: boolean) => {
+        try {
+            let existingHeaders = this.event.node.res.getHeaders();
+            let existingValue = existingHeaders[key.toLowerCase()];
+
+            // we have the this.response.header for compatibility with nextJS
+            if (existingValue === undefined) {
+                this.event.node.res.setHeader(key, value);
+            } else if (allowDuplicateKey) {
+                this.event.node.res.setHeader(key, existingValue + ", " + value);
+            } else {
+                // we overwrite the current one with the new one
+                this.event.node.res.setHeader(key, value);
+            }
+        } catch (err) {
+            throw new Error("Error while setting header with key: " + key + " and value: " + value);
+        }
+    };
+
+    setCookie = (
+        key: string,
+        value: string,
+        domain: string | undefined,
+        secure: boolean,
+        httpOnly: boolean,
+        expires: number,
+        path: string,
+        sameSite: "strict" | "lax" | "none"
+    ) => {
+        setCookie(this.event, key, value, {
+            secure,
+            sameSite,
+            httpOnly,
+            expires: new Date(expires),
+            domain,
+            path,
+        });
+    };
+
+    /**
+     * @param {number} statusCode
+     */
+    setStatusCode = (statusCode: number) => {
+        if (this.event.node.res.writable) {
+            this.statusCode = statusCode;
+        }
+    };
+
+    sendJSONResponse = (content: any) => {
+        if (this.event.node.res.writable) {
+            this.event.node.res.setHeader("Content-Type", "application/json");
+            this.event.node.res.statusCode = this.statusCode;
+            send(this.event, content);
+        }
+    };
+}
+
+export interface SessionEvent extends H3Event {
+    context: {
+        session?: SessionContainerInterface;
+    };
+}
+
+export const middleware = eventHandler(async (event) => {
+    const supertokens = SuperTokens.getInstanceOrThrowError();
+    const request = new H3Request(event);
+    const response = new H3Response(event);
+    try {
+        await supertokens.middleware(request, response);
```

Comment:
h3 middlewares seem to be compatible with express' middleware. Therefore, I think you can just copy over the logic from the [express framework](https://github.com/supertokens/supertokens-node/blob/master/lib/ts/framework/express/framework.ts#L152) (but of course, use the `H3Request` and `H3Response` wrapper instead of `ExpressRequest` and `ExpressResponse`.

The key point here is that `await supertokens.middleware(request, response)` returns a boolean. If it's `false`, it means that supertokens didn't handle the request and you need to call the `next` function.